tests/common/grpc_test.go

```diff
@@ -12,95 +12,102 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build !cluster_proxy
-
-package e2e
+package common
 
 import (
 	"context"
 	"fmt"
-	"net/url"
-	"strings"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
-	"go.etcd.io/etcd/pkg/v3/expect"
 	"go.etcd.io/etcd/tests/v3/framework/config"
-	"go.etcd.io/etcd/tests/v3/framework/e2e"
+	intf "go.etcd.io/etcd/tests/v3/framework/interfaces"
 	"go.etcd.io/etcd/tests/v3/framework/testutils"
 )
 
+type clientSocket int
+
+const (
+	clientSocketTCP clientSocket = iota
+	clientSocketUnix
+)
+
 func TestAuthority(t *testing.T) {
+	testRunner.BeforeTest(t)
 	tcs := []struct {
```

Comment:
I based the common test cases on the e2e matrix, which is a superset of the integration matrix. 
Integration originally did not cover the “insecure TLS” scenarios (AutoTLS), but after the migration, we can unify coverage across runners so both validate the same authority matrix.